stress_test/controller.py

```diff
@@ -0,0 +1,349 @@
+        """
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            if check_controller_status(ssh_client) == '1':
+                return
+
+        raise Exception('Controller failed to start. '
+                        'Status check returned 0 after trying for {0} seconds.'.
+                        format(float(interval_ms) / 1000))
+
+
+
+class ODL(Controller):
+    def __init__(self,ctrl_base_dir,test_config):
+        Controller.__init__(self,test_config)
+
+        self.stat_period_ms = test_config['controller_statistics_period_ms']
+
+        if 'controller_flowmods_conf_handler' in test_config:
+            self.flowmods_conf_handler= ctrl_base_dir + test_config['controller_flowmods_conf_handler']
+
+        if 'controller_statistics_handler' in test_config:
+            self.statistics_handler = ctrl_base_dir + test_config['controller_statistics_handler']
+
+        if 'controller_persistent_handler' in test_config:
+            self.persistent_handler = ctrl_base_dir + test_config['controller_persistent_handler']
+
+
+    def controller_pre_actions(self, controller_ssh_client, controller_cpus):
+        """ Performs all necessary actions before starting a test. Pre actions
+        are 1) rebuild_controller 2) check_for_active_controller
+        3) generate_controller_xml_files
+
+        :param controller_ssh_client: paramiko.SSHClient object
+        :param controller_cpus: number of cpus returned by create_cpu_shares() and
+        allocated for controller
+        :type controller_ssh_client: paramiko.SSHClient
+        :type controller_cpus: str
+        """
+        if self.controller_rebuild:
+            logging.info('[controller_pre_actions] building controller')
+            rebuild_controller(self.build_handler, controller_ssh_client)
+
+        logging.info('[controller_pre_actions] checking for other active '
+                     'controllers')
+        check_for_active_controller(controller_ssh_client)
+        logging.info('[controller_pre_actions] starting and stopping controller'
+                     ' to generate xml files')
+
+        self.generate_controller_xml_files(controller_ssh_client, controller_cpus)
+        if self.persistent_handler != '':
+            change_persistent_controller(
+                self.persistent_handler,
+                controller_ssh_client)
+
+
+    def generate_controller_xml_files(self, ssh_client, controller_cpus):
+        """ Starts and then stops the controller to trigger the generation of
+        controller's XML files.
+
+        :param ssh_client : SSH client provided by paramiko to run the command
+        :param controller_cpus: number of cpus returned by create_cpu_shares() and
+        allocated for controller
+        :type ssh_client: paramiko.SSHClient
+        :type controller_cpus: str
+        """
+        logging.info('[generate_controller_xml_files] Starting controller')
```

Comment:
OK, this message is useless, since the "Starting controller" information will be printed within the `start` method